utils.js

```diff
@@ -22,6 +22,20 @@ const { Gio } = imports.gi;
 const ExtensionUtils = imports.misc.extensionUtils;
 const PanelExtension = ExtensionUtils.getCurrentExtension();
 
+function getSettings() {
+    const dir = PanelExtension.dir.get_child('schemas').get_path();
+    const source = Gio.SettingsSchemaSource.new_from_directory(dir,
+        Gio.SettingsSchemaSource.get_default(), false);
+    if (!source)
+        throw new Error('Error retrieving settings schema source');
+
+    const schema = source.lookup('com.endlessm.panel-extension', false);
+    if (!schema)
+        throw new Error('Schema missing');
+
+    return new Gio.Settings({ settings_schema: schema });
+}
+
```

Comment:
I don't think this is needed. You can simply call `ExtensionUtils.getSettings()`

Reply:
Ok, will try.

Reply:
Done.